packages/core/upload/server/services/api-upload-folder.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+const { isNil, get } = require('lodash/fp');
+const { getService } = require('../utils');
+
+const folderModel = 'plugin::upload.folder';
+const API_UPLOAD_FOLDER_BASE_NAME = 'Uploads';
+
+const getStore = () => strapi.store({ type: 'plugin', name: 'upload', key: 'api-folder' });
+
+const createApiUploadFolder = async () => {
+  let name = API_UPLOAD_FOLDER_BASE_NAME;
+  const folderService = getService('folder');
+
+  let exists = true;
+  let index = 1;
+  while (exists) {
+    exists = await folderService.exists({ name, parent: null }); // TODO: check if works
+    if (exists) {
+      name = `${API_UPLOAD_FOLDER_BASE_NAME} (${index})`;
```

Comment:
Do I understand it right: in case a folder with the name `API_UPLOAD_FOLDER_BASE_NAME` already exists you create a new one and append `(index)`?

Reply:
yes :)